ibridgesgui/browser.py

```diff
@@ -488,3 +485,41 @@ def _metadata_edits(self, operation):
                 elif operation == "delete":
                     meta.delete(new_key, new_val, new_units)
                 self._fill_metadata_tab(irods_path)
+
+    def _fs_select(self, path_select):
+        """Retrieve the path (file or folder) from a QFileDialog
+           path_select: PyQt6.QtWidgets.QFileDialog.getExistingDirectory
+                        PyQt6.QtWidgets.QFileDialog.getOpenFileName
+
+        """
+        yes_button = PyQt6.QtWidgets.QMessageBox.StandardButton.Yes
+        no_button = PyQt6.QtWidgets.QMessageBox.StandardButton.No
+        if isinstance(path_select, tuple):
+            path = path_select[0]
+            reply = PyQt6.QtWidgets.QMessageBox.question(self,
+                                     'Upload File', path,
+                                     yes_button | no_button, no_button)
+        else:
+            path = path_select
+            reply = PyQt6.QtWidgets.QMessageBox.question(self,
+                                     'Upload Folder', path,
+                                     yes_button | no_button, no_button)
+
+        if reply == yes_button and path != '':
+            return Path(path)
+
+    def _upload(self, source):
+        """Uploads data to path in inputPath"""
+        overwrite = self.overwrite.isChecked()
```

Comment:
This is okay for now, but I would make it something that is set when it actually tries to overwrite.